neqo-transport/src/connection/tests/datagram.rs

```diff
@@ -58,12 +58,12 @@ fn datagram_disabled_both() {
     assert_eq!(client.max_datagram_size(), Err(Error::NotAvailable));
     assert_eq!(server.max_datagram_size(), Err(Error::NotAvailable));
     assert_eq!(
-        client.send_datagram(DATA_SMALLER_THAN_MTU, None),
+        client.send_datagram(DATA_SMALLER_THAN_MTU.to_vec(), None),
```

Comment:
if this is possible with Rust, can we add this `to_vec` call to the definition of the `const`, rather than to each use? (Ditto for the other `to_vec` calls.)

Reply:
Long story short, I don't see an easy way around the `.to_vec()` calls.

Long story:

`Vec` is heap allocated, and thus can not be instantiated as a `const`, in other words, it can only be instantiated at runtime.

We could use a `lazy_static`, but then we would need to call `.clone()` everywhere, which I think adds the same amount of noise as calling `.to_vec()` everywhere.

We could change `send_datagram`:

``` diff
-    pub fn send_datagram(&mut self, buf: Vec<u8>, id: impl Into<DatagramTracking>) -> Res<()> {
+    pub fn send_datagram(&mut self, buf: impl Into<Vec<u8>>, id: impl Into<DatagramTracking>) -> Res<()> {
```

But this makes the allocation within `send_datagram` implicit to the call-site, thus once again hiding the fact that we would allocate on each datagram when called with a `&[u8]`.

Reply:
Thanks for the explanation.